Simplenote/Classes/SPNoteListViewController+Extensions.swift

```diff
@@ -622,6 +705,40 @@ extension SPNoteListViewController {
             notesListController.object(at: indexPath) as? Note
         }
     }
+
+    open override func setEditing(_ editing: Bool, animated: Bool) {
+        super.setEditing(editing, animated: animated)
+        refreshEditButtonTitle()
+        
+        tableView.deselectAllRows(inSection: 0, animated: false)
+        updateNavigationBar()
+
+        tableView.allowsMultipleSelection = editing
+        toggleMultiSelectEditingTo(editing)
+
+        configureNavigationToolbarButton()
+        setListViewTitle()
+    }
+
+    func toggleMultiSelectEditingTo(_ editing: Bool) {
+        guard let noteCells = tableView.visibleCells as? [SPNoteTableViewCell] else {
```

Comment:
Beware this could be a problem, under certain conditions. `visibleCells` returns only the reusable cells that are onscreen, which **not necessarily** will always be all of the cells in the table. ([Ref. here](https://stackoverflow.com/questions/39458544/uitableview-visiblecells-is-not-returning-all-cells)).

I've tested this PR in a large account (+500 notes), and appears to be working fine. But to be safe, we should consider:

1. Setting the `multiEdit` toggle in `configureCell`
2. Switch to `reloadRowsAtIndexPaths` (this could be an API in the UITableView extension)

WDYT?

Reply:
I had it setup that way before, but the problem with reloading all of the cells is that is changes the animation.  As opposed to the checkboxes animating in from the side, the entire cell is replaced which could be fine, but isn't what I would expect.  

I wonder if a compromise situation could be made where in configure we set the multi edit mode for the cell, but still animate the currently visible cells.  That should cover them all as new ones will dequeue in the correct mode while still giving us the slide out animation.

WDYT?